R/setkey.R

```diff
@@ -177,7 +177,14 @@ forderv <- function(x, by=seq_along(x), retGrp=FALSE, sort=TRUE, order=1L, na.la
   } else {
     if (!length(x)) return(integer(0)) # to be consistent with base::order. this'll make sure forderv(NULL) will result in error
                        # (as base does) but forderv(data.table(NULL)) and forderv(list()) will return integer(0))
-    if (is.character(by)) by=chmatch(by, names(x))
+    if (is.character(by)) {
+      w = chmatch(by, names(x))
+      if (anyNA(w)) stop("'by' contains '",by[is.na(w)][1],"' which is not a column name")
```

Comment:
heads up that `anyNA` is R 3.1, in case we decide to keep the 3.0 dependency

Reply:
Here changing to `any(is.na(.))` won't hurt much, I don't think grouping by million of columns would be useful.

Reply:
Comment for completeness ... yep good point, we're now >= R 3.1 for other reasons, so can use `anyNA`